src/modules/systemlib/battery.cpp

```diff
@@ -239,12 +230,8 @@ Battery::computeScale()
 	// reusing capacity calculation to get single cell voltage before drop
 	const float bat_v = _param_v_empty.get() + (voltage_range * _remaining_voltage);
 
-	_scale = _param_v_full.get() / bat_v;
-
-	if (_scale > 1.3f) { // Allow at most 30% compensation
-		_scale = 1.3f;
+	// Allow at most 30% compensation
+	// Shouldn't ever be more than the power at full battery
+	_scale = math::constrain(_param_v_full.get() / bat_v, 1.0f, 1.3f);
```

Comment:
Constrain already tends to hide problems. The entire thing probably needs some thought from the perspective of defensive programming. The FW controllers in L1 check every single input. https://github.com/PX4/ecl/blob/master/attitude_fw/ecl_pitch_controller.cpp#L92
Should every interface be this paranoid?